samples/extensions/subgroups_operations/subgroups_operations.h

```diff
@@ -0,0 +1,134 @@
+/* Copyright (c) 2023, Mobica Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 the "License";
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "api_vulkan_sample.h"
+
+struct TextureQuadVertex
+{
+	glm::vec3 pos;
+	glm::vec2 uv;
+};
+
+class SubgroupsOperations : public ApiVulkanSample
+{
+  public:
+	SubgroupsOperations();
+	~SubgroupsOperations();
+
+	bool prepare(vkb::Platform &platform) override;
+	void request_gpu_features(vkb::PhysicalDevice &gpu) override;
+	void build_command_buffers() override;
+	void render(float delta_time) override;
+	bool resize(const uint32_t width, const uint32_t height) override;
+	void on_update_ui_overlay(vkb::Drawer &drawer) override;
+
+	void draw();
+	void load_assets();
+
+	void prepare_graphics();
+	void generate_quad();
+	void setup_descriptor_pool();
+	void setup_descriptor_set_layout();
+	void setup_descriptor_set();
+	void setup_pipelines();
+	void create_semaphore();
+	void prepare_uniform_buffers();
+	void update_uniform_buffers();
+
+	void prepare_compute();
+	void create_compute_queue();
+	void create_compute_command_pool();
+	void create_compute_command_buffer();
+	void create_compute_descriptor_set_layout();
+	void create_compute_descriptor_set();
+
+	void preapre_compute_pipeline_layout();
+	void prepare_compute_pipeline();
+
+	void build_compute_command_buffer();
+
+	struct GuiSettings
+	{
+		int32_t                         selected_filtr = 0;
+		static std::vector<std::string> init_filters_name()
+		{
+			std::vector<std::string> filtrs = {
```

Comment:
typo. Should be std::vector<std::string> filters

Reply:
out to date